

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Getting Started &mdash; Multi Omic Graph Diagnosis (MOGDx)</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />
      <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
      <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
      <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
      <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css?v=4fa983c6" />
      <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=9eb32ce0"></script>
      <script src="_static/doctools.js?v=9a2dae69"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="_static/clipboard.min.js?v=a7894cd8"></script>
      <script src="_static/copybutton.js?v=f281be69"></script>
      <script src="_static/scripts/sphinx-book-theme.js"></script>
      <script>let toggleHintShow = 'Click to show';</script>
      <script>let toggleHintHide = 'Click to hide';</script>
      <script>let toggleOpenOnPrint = 'true';</script>
      <script src="_static/togglebutton.js?v=4a39c7ea"></script>
      <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
      <script src="_static/design-tabs.js?v=f930bc37"></script>
      <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
      <script async="async" src="_static/sphinx-thebe.js?v=c100c467"></script>
      <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
      <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Processing in R" href="Processing_in_R.html" />
    <link rel="prev" title="Multi-Omic Graph Diagnosis (MOGDx)" href="intro.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="intro.html">
            
              <img src="_static/mogdx.png" class="logo" alt="Logo"/>
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Introduction</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting Started</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Data Processing</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Processing_in_R.html">Processing in R</a></li>
<li class="toctree-l1"><a class="reference internal" href="Processing_in_Python.html">Processing in Python</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">MOGDx Deployment</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Executing_MOGDx.html">Executing MOGDx</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">MOGDx Model Types</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="MOGDx/Notebooks/MOGDx.html">MOGDx</a></li>
<li class="toctree-l1"><a class="reference internal" href="MOGDx/Notebooks/MOGDx_PNet.html">MOGDx &amp; PNet</a></li>
<li class="toctree-l1"><a class="reference internal" href="MOGDx/Notebooks/MOGDx_NetGen.html">MOGDx Network Generation</a></li>
<li class="toctree-l1"><a class="reference internal" href="MOGDx/Notebooks/MOGDx_Free.html">MOGDx Free</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Model Development</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="Training_MOGDx.html">MOGDx Main Functions and Classes</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Contact Details</a></li>
<li class="toctree-l1"><a class="reference internal" href="requirements.html">Requirements</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="intro.html">Project name not set</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="intro.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Getting Started</li>
      <li class="wy-breadcrumbs-aside">
              <a href="https://github.com/Barry8197/MOGDxBook/blob/master/_build/html/_sources/GettingStarted.md" class="fa fa-github"> Edit on GitHub</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section class="tex2jax_ignore mathjax_ignore" id="getting-started">
<h1>Getting Started<a class="headerlink" href="#getting-started" title="Link to this heading"></a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Link to this heading"></a></h2>
<p>Multi-omic Graph Diagnosis (MOGDx) is a tool for the integration of omic data and classification of heterogeneous diseases. MOGDx exploits a Patient Similarity Network (PSN) framework to integrate omic data using Similarity Network Fusion (SNF) <span id="id1">[<a class="reference internal" href="#id5" title="Bo Wang, Aziz M. Mezlini, Feyyaz Demir, Marc Fiume, Zhuowen Tu, Michael Brudno, Benjamin Haibe-Kains, and Anna Goldenberg. Similarity network fusion for aggregating data types on a genomic scale. Nature Methods, 11(3):333–337, March 2014. Number: 3 Publisher: Nature Publishing Group. URL: https://www.nature.com/articles/nmeth.2810 (visited on 2024-02-12), doi:10.1038/nmeth.2810.">Wang <em>et al.</em>, 2014</a>]</span>. A single PSN is built per modality. The PSN is built using the most informative features of that modality. The most informative features are found by performing a contrastive analysis between classification targets. For example, in mRNA, differentially expressed genes will be used to construct the PSN. Where suitable, Pearson correlation, otherwise Euclidean distance is measured between these informative features and the network is constructed using the K Nearest Neighbours (KNN) algorithm. SNF is used to combine individual PSN’s into a single network. The fused PSN and the omic datasets are input into the Graph Convultional Network with Multi Modal Encoder, the architecture of which is shown in below. Each omic measure is compressed using a two layer encoder. The compressed encoded layer of each modality is then decoded to a shared latent space using mean pooling. This encourages each modality to learn the same latent space. The shared latent space is the node feature matrix, required for training the GCN, with each row forming a node feature vector. Classification is performed on the fused network using the Graph Convolutional Network (GCN) deep learning algorithm <span id="id2">[<a class="reference internal" href="Training_MOGDx.html#id12" title="Thomas N. Kipf and Max Welling. Semi-Supervised Classification with Graph Convolutional Networks. arXiv, February 2017. arXiv:1609.02907 [cs, stat]. URL: http://arxiv.org/abs/1609.02907 (visited on 2022-09-26).">Kipf and Welling, 2017</a>]</span>.</p>
<p>GCN is a novel paradigm for learning from both network structure and node features. Heterogeneity in diseases confounds clinical trials, treatments, genetic association and more. Accurate stratification of these diseases is therefore critical to optimize treatment strategies for patients with heterogeneous diseases. Previous research has shown that accurate classification of heterogenous diseases has been achieved by integrating and classifying multi-omic data <span id="id3">[<a class="reference internal" href="#id7" title="Xiao Li, Jie Ma, Ling Leng, Mingfei Han, Mansheng Li, Fuchu He, and Yunping Zhu. MoGCN: A Multi-Omics Integration Method Based on Graph Convolutional Network for Cancer Subtype Analysis. Frontiers in Genetics, 2022. URL: https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2022.806842 (visited on 2024-02-12).">Li <em>et al.</em>, 2022</a>, <a class="reference internal" href="#id6" title="Shraddha Pai, Shirley Hui, Ruth Isserlin, Muhammad A Shah, Hussam Kaka, and Gary D Bader. netDx: interpretable patient classification using integrated patient similarity networks. Molecular Systems Biology, 15(3):e8497, March 2019. URL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6423721/ (visited on 2024-02-12), doi:10.15252/msb.20188497.">Pai <em>et al.</em>, 2019</a>, <a class="reference internal" href="#id8" title="Tongxin Wang, Wei Shao, Zhi Huang, Haixu Tang, Jie Zhang, Zhengming Ding, and Kun Huang. MOGONET integrates multi-omics data using graph convolutional networks allowing patient classification and biomarker identification. Nature Communications, 12(1):3445, June 2021. Number: 1 Publisher: Nature Publishing Group. URL: https://www.nature.com/articles/s41467-021-23774-w (visited on 2024-02-12), doi:10.1038/s41467-021-23774-w.">Wang <em>et al.</em>, 2021</a>]</span>. MOGDx improves upon this research. The advantages of MOGDx is that it can handle both a variable number of modalities and missing patient data in one or more modalities. Performance of MOGDx was benchmarked on the BRCA TCGA dataset with competitive performance compared to its counterparts. In summary, MOGDx combines patient similarity network integration with graph neural network learning for accurate disease classification.</p>
</section>
<section id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Link to this heading"></a></h2>
<p>A working version of R and Python is required. R version 4.2.2 and Python version 3.9.2 was used to obtain all results.</p>
</section>
<section id="data-download">
<h2>Data Download<a class="headerlink" href="#data-download" title="Link to this heading"></a></h2>
<p>Create a folder called <code class="docutils literal notranslate"><span class="pre">data</span></code> and a folder for each dataset with naming convention ‘TCGA-’ e.g. ‘TCGA-BRCA’ inside this folder.</p>
<p>Use the R script <code class="docutils literal notranslate"><span class="pre">data_download.R</span></code> to download all data changing the project to BRCA/LGG/KICH/KIRC/KICH</p>
<p>Note : To create the KIPAN dataset, the KIRC, KICP and KICH datasets have to be combined. This can be achieved by copying the downloaded files
from all three seperate datasets into a single dataset called KIPAN keeping the same naming structures. A column named ‘subtype’ specifying which dataset
the patient came from needs to be created in the Meta data file. A basic knowledge of R is required for this.</p>
</section>
<section id="accessing-the-data-in-python">
<h2>Accessing the data in Python<a class="headerlink" href="#accessing-the-data-in-python" title="Link to this heading"></a></h2>
<p>The downloaded data in its raw format will be in <code class="docutils literal notranslate"><span class="pre">.rda</span></code> format. It will be required to convert from this format to <code class="docutils literal notranslate"><span class="pre">.pkl</span></code> for processing in python.</p>
<p>It is recommended to use the <a class="reference external" href="https://pypi.org/project/pyreadr/">pyreadr</a> python package to do so and saving the preprocessed file with naming convention <code class="docutils literal notranslate"><span class="pre">{modality}_preprocessed.pkl</span></code></p>
</section>
<section id="citations">
<h2>Citations<a class="headerlink" href="#citations" title="Link to this heading"></a></h2>
<div class="docutils container" id="id4">
<div role="list" class="citation-list">
<div class="citation" id="id10" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id2">KW17</a><span class="fn-bracket">]</span></span>
<p>Thomas N. Kipf and Max Welling. <em>Semi-Supervised Classification with Graph Convolutional Networks</em>. arXiv, February 2017. arXiv:1609.02907 [cs, stat]. URL: <a class="reference external" href="http://arxiv.org/abs/1609.02907">http://arxiv.org/abs/1609.02907</a> (visited on 2022-09-26).</p>
</div>
<div class="citation" id="id7" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id3">LML+22</a><span class="fn-bracket">]</span></span>
<p>Xiao Li, Jie Ma, Ling Leng, Mingfei Han, Mansheng Li, Fuchu He, and Yunping Zhu. MoGCN: A Multi-Omics Integration Method Based on Graph Convolutional Network for Cancer Subtype Analysis. <em>Frontiers in Genetics</em>, 2022. URL: <a class="reference external" href="https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2022.806842">https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2022.806842</a> (visited on 2024-02-12).</p>
</div>
<div class="citation" id="id6" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id3">PHI+19</a><span class="fn-bracket">]</span></span>
<p>Shraddha Pai, Shirley Hui, Ruth Isserlin, Muhammad A Shah, Hussam Kaka, and Gary D Bader. netDx: interpretable patient classification using integrated patient similarity networks. <em>Molecular Systems Biology</em>, 15(3):e8497, March 2019. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6423721/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6423721/</a> (visited on 2024-02-12), <a class="reference external" href="https://doi.org/10.15252/msb.20188497">doi:10.15252/msb.20188497</a>.</p>
</div>
<div class="citation" id="id5" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id1">WMD+14</a><span class="fn-bracket">]</span></span>
<p>Bo Wang, Aziz M. Mezlini, Feyyaz Demir, Marc Fiume, Zhuowen Tu, Michael Brudno, Benjamin Haibe-Kains, and Anna Goldenberg. Similarity network fusion for aggregating data types on a genomic scale. <em>Nature Methods</em>, 11(3):333–337, March 2014. Number: 3 Publisher: Nature Publishing Group. URL: <a class="reference external" href="https://www.nature.com/articles/nmeth.2810">https://www.nature.com/articles/nmeth.2810</a> (visited on 2024-02-12), <a class="reference external" href="https://doi.org/10.1038/nmeth.2810">doi:10.1038/nmeth.2810</a>.</p>
</div>
<div class="citation" id="id8" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id3">WSH+21</a><span class="fn-bracket">]</span></span>
<p>Tongxin Wang, Wei Shao, Zhi Huang, Haixu Tang, Jie Zhang, Zhengming Ding, and Kun Huang. MOGONET integrates multi-omics data using graph convolutional networks allowing patient classification and biomarker identification. <em>Nature Communications</em>, 12(1):3445, June 2021. Number: 1 Publisher: Nature Publishing Group. URL: <a class="reference external" href="https://www.nature.com/articles/s41467-021-23774-w">https://www.nature.com/articles/s41467-021-23774-w</a> (visited on 2024-02-12), <a class="reference external" href="https://doi.org/10.1038/s41467-021-23774-w">doi:10.1038/s41467-021-23774-w</a>.</p>
</div>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="intro.html" class="btn btn-neutral float-left" title="Multi-Omic Graph Diagnosis (MOGDx)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="Processing_in_R.html" class="btn btn-neutral float-right" title="Processing in R" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>
    <!-- Theme Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-Z09VV2YFE0"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-Z09VV2YFE0', {
          'anonymize_ip': false,
      });
    </script> 

</body>
</html>